azure/errors.go

```diff
@@ -0,0 +1,9 @@
+package azure
+
+const (
+	publishSettingsConfigurationError = "PublishSettingsFilePath is set. Consequently ManagementCertificatePath and SubscriptionId must not be set."
+
+	managementCertificateConfigurationError = "ManagementCertificatePath is set. Consequently SubscriptionId must also be set, and PublishSettingsFilePath must not be set."
```

Comment:
How do you know that ManagementCertificatePath  is set?
Condition was: len(config.ManagementCertificatePath) == 0 || len(config.SubscriptionId) == 0
They both can be null at this point.
Mb we should say something like "Both ManagementCertificatePath and SubscriptionId should be set, and PublishSettingsFilePath must not be set."